### PR TITLE
feat: add table

### PR DESCRIPTION
**Pull request by jesperwiner**

### CODE REVIEW

File: src/Atoms/Table/Table.types.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export type Props = {
+  children: React.ReactNode;
+  tableLayout?: string;
+  width?: number | string;
```
Comment (jauniuse) [inline src/Atoms/Table/Table.types.ts]: probably `number` is not a valid css width value

File: src/Atoms/Table/Table.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+let tableData: (string | number)[][] = [];
+const multiplyTableWith = 5;
+
+for (let i = 0; i < multiplyTableWith; i += 1) {
+  tableData = tableData.concat(data);
+}
+
+/* eslint-disable react/no-array-index-key */
+storiesOf('Atoms | Table', module)
+  .add('Table', () => {
```
Comment (shmidt-i) [inline src/Atoms/Table/Table.stories.tsx]: If you are using typography in the story, story needs to be named as 'Integration: ...'

File: src/Atoms/Table/Table.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+              <Td>-</Td>
+              <Td>-</Td>
+              <Td>-</Td>
+            </Tr>
+          </Tfoot>
+        </Table>
+      </Typography>
+    );
+  })
+  .add('Table with stickyHeader', () => {
```
Comment (shmidt-i) [inline src/Atoms/Table/Table.stories.tsx]: "Integration: " here as well

File: src/Atoms/Table/Table.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import styled from 'styled-components';
+import { TableComponent, Props } from './Table.types';
+
+const StyledTable = styled.table<Props>`
+  width: ${p => p.width};
```
Comment (shmidt-i) [inline src/Atoms/Table/Table.tsx]:
```
width: ${p => p.width}px;
```
Forgot px?

File: src/Atoms/Table/Table.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  table-layout: ${p => (p.tableLayout ? 'fixed' : 'auto')};
+  border-collapse: collapse;
+`;
+
+export const Table: TableComponent = ({ children, width }) => (
+  <StyledTable width={width}>{children}</StyledTable>
+);
+
+Table.defaultProps = {
```
Comment (shmidt-i) [inline src/Atoms/Table/Table.tsx]: better to do defaults inside function

File: src/Atoms/Table/Table.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  tableLayout: 'fixed',
```
Comment (shmidt-i) [inline src/Atoms/Table/Table.tsx]: You are not passing tableLayout to StyledTable, is it planned? ;)

File: src/Atoms/Table/Tbody/Tbody.types.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+export type Props = {
+  children: React.ReactNode;
```
Comment (shmidt-i) [inline src/Atoms/Table/Tbody/Tbody.types.ts]: If only children are in your Props types, you can skip it cause `children` are already in React.FunctionComponent definition

File: src/Atoms/Table/Td/Td.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  :last-child {
+    padding-right: 0;
+  }
+`;
+
+export const Td: TdComponent = ({ textAlign, children }) => (
+  <StyledTd textAlign={textAlign}>{children}</StyledTd>
+);
+
+Td.defaultProps = {
```
Comment (shmidt-i) [inline src/Atoms/Table/Td/Td.tsx]: Defaults in function please

File: src/Atoms/Table/Th/Th.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  }
+`;
+
+export const Th: ThComponent = ({ width, textAlign, children }) => (
+  <StyledTh width={width} textAlign={textAlign}>
+    {children}
+  </StyledTh>
+);
+
+Th.defaultProps = {
```
Comment (shmidt-i) [inline src/Atoms/Table/Th/Th.tsx]: defaults in fn

File: src/Atoms/Table/Tr/Tr.tsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    border-top: 1px solid ${p => p.theme.color.divider};
+  }
+  overflow: 'hidden';
+`;
+
+export const Tr: TrComponent = ({ textAlign, children }) => (
+  <StyledTr textAlign={textAlign}>{children}</StyledTr>
+);
+
+Tr.defaultProps = {
```
Comment (shmidt-i) [inline src/Atoms/Table/Tr/Tr.tsx]: defaults in fn

File: src/Atoms/Table/Table.tsx (hunk 6 above)
Comment (jesperwiner) [reply src/Atoms/Table/Table.tsx]: soon (tm)

File: src/Atoms/Table/Table.types.ts (hunk 1 above)
Comment (jesperwiner) [reply src/Atoms/Table/Table.types.ts]: Fixed

File: src/Atoms/Table/Table.stories.tsx (hunk 2 above)
Comment (jesperwiner) [reply src/Atoms/Table/Table.stories.tsx]: Fixed

File: src/Atoms/Table/Table.stories.tsx (hunk 3 above)
Comment (jesperwiner) [reply src/Atoms/Table/Table.stories.tsx]: Fixed

File: src/Atoms/Table/Table.tsx (hunk 4 above)
Comment (jesperwiner) [reply src/Atoms/Table/Table.tsx]: No, we want to use both % and px

File: src/Atoms/Table/Table.tsx (hunk 5 above)
Comment (jesperwiner) [reply src/Atoms/Table/Table.tsx]: Fixed

File: src/Atoms/Table/Table.tsx (hunk 6 above)
Comment (jesperwiner) [reply src/Atoms/Table/Table.tsx]: Fixed

File: src/Atoms/Table/Tbody/Tbody.types.ts (hunk 7 above)
Comment (jesperwiner) [reply src/Atoms/Table/Tbody/Tbody.types.ts]: Fixed

File: src/Atoms/Table/Td/Td.tsx (hunk 8 above)
Comment (jesperwiner) [reply src/Atoms/Table/Td/Td.tsx]: Fixed

File: src/Atoms/Table/Th/Th.tsx (hunk 9 above)
Comment (jesperwiner) [reply src/Atoms/Table/Th/Th.tsx]: Fixed

File: src/Atoms/Table/Tr/Tr.tsx (hunk 10 above)
Comment (jesperwiner) [reply src/Atoms/Table/Tr/Tr.tsx]: Fixed

**Pull request merged**